stellargraph/data/explorer.py

```diff
@@ -849,3 +854,342 @@ def _check_neighbourhood_sizes(self, in_size, out_size):
+
+                        if (not move_backwards) and (
+                            not move_forwards
+                        ):  # if dead ends reached in both direction, stop walking.
+                            break
+                    walks.append(walk)
+
+        return walks
+
+    def _check_temporal_parameters(self, bidirectional, edge_time_label):
+        """
+            Checks that the parameter values are valid or raises ValueError exceptions with a message indicating the
+            parameter (the first one encountered in the checks) with invalid value.
+    
+            Args:
+                bidirectional: <False or True> Indicates whether the temporal walk is in forward direction only or extends in both directions.
+                edge_time_label: <string> Label of the edge weight property.
+           """
+
+        if type(bidirectional) != bool:
+            self._raise_error(
+                "Parameter bidirectional has to be either False (temporal random walks forward in times) or True (temporal random walks extending in both directions)."
+            )
+
+        if not isinstance(edge_time_label, str):
+            self._raise_error("The edge time property label has to be of type string")
+
+        "Check that all edges have the time label specified by 'edge_time_label' and all the time values positive real values."
+        for node in self.graph.nodes():
+            for neighbor in self.neighbors(node):
+                for k, v in self.graph[node][neighbor].items():
+                    if edge_time_label in v:
+                        t = v.get(edge_time_label)
+                        if t is None or np.isnan(t) or t == np.inf:
+                            self._raise_error(
+                                "Missing or invalid time ({}) between ({}) and ({}).".format(
+                                    t, node, neighbor
+                                )
+                            )
+                        if not isinstance(t, (int, float)):
+                            self._raise_error(
+                                "Timestamp between nodes ({}) and ({}) is not numeric ({}).".format(
+                                    node, neighbor, t
+                                )
+                            )
+                        if t < 0:  # check if edge has a negative timestamp
+                            self._raise_error(
+                                "Timestamp between nodes ({}) and ({}) is negative ({}).".format(
+                                    node, neighbor, t
+                                )
+                            )
+                    else:
+                        self._raise_error(
+                            "Invalid time label ({}) between ({}) and ({}).".format(
+                                edge_time_label, k, v
+                            )
+                        )
+
+
+class TemporalBiasedRandomWalk(GraphWalk):
```

Comment:
I think once the `Uniform` version above has its main chunks of logic split out into smaller functions, we could probably think more about how to share more code between these two classes, unless you had explicit reasons for not wanting to do that and keep them entirely separate?

Reply:
Done. Now there is only `TemporalRandomWalk` class.  The run method takes a `step_type` value to choose between `uniform_random` or `biased_random` step (forward or backward) in time. Biased walk here means exponential random but could be any other distribution. The authors of this work proposed this exponential and linear type of forward step. Linear didn't make much sense so left it for now.